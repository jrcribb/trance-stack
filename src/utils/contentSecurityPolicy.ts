import { getCSP, SELF, UNSAFE_INLINE, BLOB, DATA, NONE } from 'csp-header';

/**
 * @name contentSecurityPolicy
 * @description A better way to set CSP directives
 */
export const contentSecurityPolicy = (isDevelopment: boolean): string => {
  return getCSP({
    directives: {
      'default-src': [SELF],
      'worker-src': [BLOB],
      'script-src': [
        SELF,
        UNSAFE_INLINE,
        'https://*.cookieyes.com',
        'https://cdn-cookieyes.com',
        'https://*.mixpanel.com',
        'https://*.hotjar.com',
        'https://*.googletagmanager.com',
        'https://*.google-analytics.com',
        'https://*.hotjar.io'
      ],
      'connect-src': [
        SELF,
        UNSAFE_INLINE,
        'https://*.mixpanel.com',
        'https://*.cookieyes.com',
        'https://cdn-cookieyes.com',
        'https://*.googletagmanager.com',
        'https://*.google-analytics.com',
        'wss://*.hotjar.com',
        'https://*.hotjar.com',
        'https://*.hotjar.io',
        isDevelopment ? 'ws://localhost:2222' : '',
        isDevelopment ? 'ws://localhost:8002' : ''
      ],
      'img-src': ['https:', DATA, 'http:'],
      'frame-ancestors': [NONE],
      'frame-src': [
        'https://*.mixpanel.com',
        'https://*.cookieyes.com',
        'https://cdn-cookieyes.com',
        'https://*.googletagmanager.com',
        'https://*.google-analytics.com',
        'https://*.hotjar.com',
        'https://*.hotjar.io'
      ],
      'style-src': [
        UNSAFE_INLINE,
        SELF
      ]
    }
  });
};
