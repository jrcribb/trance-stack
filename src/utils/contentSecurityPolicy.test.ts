import { describe, it } from 'vitest';
import { contentSecurityPolicy } from './contentSecurityPolicy';

describe('contentSecurityPolicy', () => {
  it('in dev mode', () => {
    expect(contentSecurityPolicy(true)).toMatchSnapshot();
  });
  it('in prod mode', () => {
    expect(contentSecurityPolicy(false)).toMatchSnapshot();
  });
});
