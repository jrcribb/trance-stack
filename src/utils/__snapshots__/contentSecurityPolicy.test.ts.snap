// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`contentSecurityPolicy > in dev mode 1`] = `"default-src 'self'; worker-src blob:; script-src 'self' 'unsafe-inline' https://*.cookieyes.com https://cdn-cookieyes.com https://*.mixpanel.com https://*.hotjar.com https://*.hotjar.io; connect-src 'self' 'unsafe-inline' https://*.mixpanel.com https://*.cookieyes.com https://cdn-cookieyes.com wss://*.hotjar.com https://*.hotjar.com https://*.hotjar.io ws://localhost:2222 ws://localhost:8002; img-src https: data: http:; frame-ancestors 'none'; frame-src https://*.mixpanel.com https://*.cookieyes.com https://cdn-cookieyes.com https://*.hotjar.com https://*.hotjar.io; style-src 'unsafe-inline' 'self';"`;

exports[`contentSecurityPolicy > in prod mode 1`] = `"default-src 'self'; worker-src blob:; script-src 'self' 'unsafe-inline' https://*.cookieyes.com https://cdn-cookieyes.com https://*.mixpanel.com https://*.hotjar.com https://*.hotjar.io; connect-src 'self' 'unsafe-inline' https://*.mixpanel.com https://*.cookieyes.com https://cdn-cookieyes.com wss://*.hotjar.com https://*.hotjar.com https://*.hotjar.io  ; img-src https: data: http:; frame-ancestors 'none'; frame-src https://*.mixpanel.com https://*.cookieyes.com https://cdn-cookieyes.com https://*.hotjar.com https://*.hotjar.io; style-src 'unsafe-inline' 'self';"`;
